csharp/src/Vista.SDK/ILocalIdBuilder.cs

```diff
@@ -28,19 +28,28 @@ public interface ILocalIdBuilder
 
     bool HasCustomTag { get; }
 
-    LocalIdBuilder WithVisVersion(string visVersion);
+    LocalIdBuilder WithVisVersion(in string visVersion);
+    LocalIdBuilder WithVisVersion(in VisVersion version);
+    LocalIdBuilder TryWithVisVersion(in VisVersion? version);
+    bool TryWithVisVersion(in string? visVersionStr, out LocalIdBuilder visVersion);
+    LocalIdBuilder WithoutVisVersion();
 
-    LocalIdBuilder WithVisVersion(VisVersion version);
+    LocalIdBuilder WithVerboseMode(in bool verboseMode);
 
-    LocalIdBuilder WithVerboseMode(bool verboseMode);
+    LocalIdBuilder WithPrimaryItem(in GmodPath item);
+    LocalIdBuilder TryWithPrimaryItem(in GmodPath? item);
+    bool TryWithPrimaryItem(in GmodPath? item, out LocalIdBuilder localIdbuilder);
```

Comment:
bool TryWithPrimaryItem(in GmodPath? item)

if(!builder.TryWithPrimaryItem(path)) throw()